src/Orleans/Providers/ProviderLoader.cs

```diff
@@ -51,7 +52,7 @@ public void LoadProviders(IDictionary<string, IProviderConfiguration> configs, I
             }
 
             // Load providers
-            ProviderTypeLoader.AddProviderTypeManager(t => typeof(TProvider).IsAssignableFrom(t), RegisterProviderType);
+            ProviderTypeLoader.AddProviderTypeManager(t => typeof(TProvider).IsAssignableFrom(t), RegisterProviderType, serviceProvider.GetRequiredService<LoadedProviderTypeLoaders>());
```

Comment:
Is it possible to just inject `LoadedProviderTypeLoaders` into the constructor instead?